modules/openid_connect/app/services/openid_connect/user_tokens/fetch_service.rb

```diff
@@ -66,31 +67,17 @@ def access_token_for(audience:)
         Success(token.access_token)
       end
 
-      ##
-      # Obtains an access token that can be used to make requests in the user's name at the
-      # remote service identified by the +audience+ parameter.
-      #
-      # The access token will always be refreshed before being returned by this method.
-      # It is advised to use this method, after learning that a remote service rejected
-      # an access token, because it was expired.
-      #
-      # A token exchange is attempted, if the provider supports OAuth 2.0 Token Exchange and a token
-      # for the target audience either can't be found or it has expired, but has no available refresh token.
-      def refreshed_access_token_for(audience:)
-        token = yield token_with_audience(audience)
-        token = yield @token_refresh.call(token)
-        Success(token.access_token)
-      end
-
       private
 
       def token_with_audience(aud)
         token = @user.oidc_user_tokens.with_audience(aud).first
         return Success(token) if token
 
-        return @token_exchange.call(aud) if @token_exchange.supported?
-
-        Failure("No token for audience '#{aud}'")
+        if @token_exchange.supported?
+          @token_exchange.call(aud)
+        else
+          Failure(@error.with(code: :no_token_for_audience, payload: "No token for audience '#{aud}'"))
```

Comment:
TIL: Apparently our Rubocop is not configured to prefer guard clauses with no questions asked. I am pretty sure that a default-rubocop would opinionate against this new code in favor of the previous guard clause.

Reply:
Yeah, I am also indifferent TBH. This was really just a TIL, not a request for change :)